docs/reference/commandline/service_create.md

```diff
@@ -678,6 +678,25 @@ appends a new placement preference after all existing placement preferences.
 `--placement-pref-rm` removes an existing placement preference that matches the
 argument.
 
+### Specify the minimum memory needed for a service (--reserve-memory)
+
+If your service needs a minimum amount of memory in order to run correctly,
+you can use `--reserve-memory` to specify that the service should reserve a
+given amount of memory on workers. The following example reserves 4GB of
+memory for the service.
+
+```bash
+$ docker service create --reserve-memory=4GB --name=too-big nginx:alpine
+```
+
+If no workers are available that meet the requirement,
+the service will remain in a pending state until the requirement can be met.
+
+This does not guarantee that Docker will not use more memory on your host than
+you want. For instance, you could create many services, the sum of whose memory
+usage could exhaust the available memory. In order to limit Docker's overall
+memory usage, use `cgroups` or other operating system utilities.
```

Comment:
We might want to recommend `--limit-memory` for this, and also contrast it to `--reserve-memory`.